lib/routemaster/api_client.rb

```diff
@@ -111,10 +111,11 @@ def _assert_uri(url)
     def _request(method, url:, body: nil, headers:, params: {})
       uri = _assert_uri(url)
       auth = auth_header(uri.host)
+      headers = [*user_agent_header, *auth, *headers].to_h
       connection.public_send(method) do |req|
         req.url uri.to_s
         req.params.merge! params
-        req.headers = headers.merge(auth)
+        req.headers = headers
```

Comment:
IMO, it'll be clearer to append another `merge`

```
req.headers['User-Agent'] = @source_peer || "Faraday v#{Faraday::VERSION}"
```

also keeps your changes minimum to this file and avoid going from hash to array to hash as is being done right now at line 114

Reply:
Feel like the current implementation is cleaner. The overheads associated going from hash to array to hash are minimal, especially in the context of a network request.